src/test/regress/sql/multi_fix_partition_shard_index_names.sql

```diff
@@ -0,0 +1,253 @@
+----------------------------------------------------
+-- multi_fix_partition_shard_index_names
+-- check the following two issues
+-- https://github.com/citusdata/citus/issues/4962
+-- https://github.com/citusdata/citus/issues/5138
```

Comment:
Note that when I follow the steps in https://github.com/citusdata/citus/issues/5138, it crashes due to the issue I brought up earlier.

So, please make sure to add one test covering it.